tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -130,3 +130,77 @@ def test_vm_snapshot_ops(
             source_csum == res_csum
         ), f"Failed: MD5 comparison between source {vm_obj.name} and cloned {res_vm_obj.name} VMs"
         res_vm_obj.stop()
+
+    @workloads
+    @pytest.mark.polarion_id("OCS-")
+    def test_vm_snapshot_pvc_clone(
+        self,
+        # setup_cnv,
+        project_factory,
+        multi_cnv_workload,
+        snapshot_factory,
+        snapshot_restore_factory,
+        cnv_workload,
+        clone_vm_workload,
+    ):
+        """
+        This test checks the clone of restored snapshot PVC created successfully
+        without data loss or corruption.
+
+        Test steps:
+        1. Create a VM with PVC/DVT
+        2. Add data to the VM and shut it down
+        3. Take a snapshot of the VM’s PVC
+        4. Deploy a new VM using the restored snapshot PVC
+        5. Clone the VM workload to create a new PVC from the restored snapshot PVC
+        6. Check data integrity in the cloned VM
+        7. Verify that the data persisted after cloning
+        8. Adding data on cloned VM
+        """
+        proj_obj = project_factory()
+        file_paths = ["/file.txt", "/new_file.txt"]
+        vm_objs_def, vm_objs_aggr, _, _ = multi_cnv_workload(
+            namespace=proj_obj.namespace
+        )
+        vm_list = vm_objs_def + vm_objs_aggr
+        log.info(f"Total VMs to process: {len(vm_list)}")
+        for index, vm_obj in enumerate(vm_list):
+            source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
+            vm_obj.stop()
```

Comment:
Why are we stopping the VM? 
The real-world use case would involve a running VM.
We should take a snapshot when the VM is running and in the data integrity check allow a range of results for accommodating the time between source_csum and the snapshot creation

Reply:
We've explored this and identified a solution using an API-driven approach. We'll leverage the openshift-python-wrapper to create and restore snapshots, which supports taking snapshots of running VMs.